Bukkit/src/main/java/com/plotsquared/bukkit/listener/PlayerEventListener.java

```diff
@@ -1068,6 +1070,34 @@ public void onInteract(PlayerInteractEvent event) {
         }
     }
 
+    // Boats can sometimes be placed on interactable blocks such as levers,
+    // see PS-175. Armor stands, minecarts and end crystals (the other entities
+    // supported by this event) don't have this issue.
+    @EventHandler(priority = EventPriority.LOW, ignoreCancelled = true)
+    public void onBoatPlace(EntityPlaceEvent event) {
+        Player player = event.getPlayer();
+        if (player == null) {
+            return;
+        }
+        Entity placed = event.getEntity();
+        if (!(placed instanceof Boat)) {
+            return;
+        }
+        BukkitPlayer pp = BukkitUtil.getPlayer(player);
+        PlotArea area = pp.getPlotAreaAbs();
+        if (area == null) {
+            return;
+        }
+        PlayerBlockEventType eventType = PlayerBlockEventType.PLACE_VEHICLE;
+        Block block = event.getBlock();
+        BlockType blockType = BukkitAdapter.asBlockType(block.getType());
+        Location location = BukkitUtil.getLocation(block.getLocation());
+        if (!PlotSquared.get().getEventDispatcher()
+            .checkPlayerBlockEvent(pp, eventType, location, blockType, true)) {
+            event.setCancelled(true);
```

Comment:
you ought to `event.setUseInteractedBlock(Event.Result.DENY);` here as well

Reply:
There's no such method for `EntityPlaceEvent`